conftest.py

```diff
@@ -23,6 +24,11 @@
 # ---- Pytest adjustments
 import pytest
 
+# Spyder imports
+from spyder.config.base import running_in_ci
+from spyder.utils.environ import (get_user_env, set_user_env,
+                                  amend_user_shell_init)
```

Comment:
This file should be as free of Spyder imports as possible to avoid possible circular imports (`pytest` does a lot of black magic so it's hard to predict if this will have repercussions in the future). There's also a message on the docstring of this file that warns about adding fixtures here.

So, I suggest to move the `restore_user_env` fixture to `spyder/app/tests/conftest.py` and then import it in `spyder/utils/tests/test_environ.py` (no changes will be required in `test_mainwindow.py`).

Reply:
Thank you for the clarification, @ccordoba12. I saw the warning message but also saw another fixture in here, so I didn't know how strict we were with that. But the concern for Spyder imports makes sense.